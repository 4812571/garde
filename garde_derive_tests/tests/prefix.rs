#[path = "./util/mod.rs"]
mod util;

#[derive(Debug, garde::Validate)]
struct Test<'a> {
    #[garde(prefix("test"))]
    field: &'a str,
}

#[test]
fn prefix_valid() {
    util::check_ok(&[Test { field: "test" }, Test { field: "test_asdf" }], &())
}

#[test]
fn prefix_invalid() {
    util::check_fail!(&[Test { field: "a" }, Test { field: "_test" }], &())
}
