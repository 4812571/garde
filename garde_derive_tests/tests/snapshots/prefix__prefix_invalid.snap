---
source: garde_derive_tests/tests/prefix.rs
expression: snapshot
---
Test {
    field: "a",
}
`field` does not begin with "test"

Test {
    field: "_test",
}
`field` does not begin with "test"


